packages/core/src/base/cacheManagerKeyv.ts

```diff
@@ -0,0 +1,277 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+"use strict";
+
+import fetch = require("minipass-fetch");
+import Keyv = require("keyv");
+import ssri = require("ssri");
+import url = require("url");
+
+import { ICacheManager } from "./cacheManager";
+
+/**
+ * Calculate a reasonable time to live for the response based on the request and response headers.
+ *
+ * @param response The response to calculate the TTL for
+ */
+const getTTL = (response: fetch.Response): number => {
+  const responseControl = response.headers
+    .get("cache-control")
+    ?.toLowerCase()
+    .trim()
+    .split(/\s*,\s*/);
+
+  if (responseControl) {
+    if (responseControl.find("private") || responseControl.find("no-store")) {
+      return 0;
+    }
+
+    const sMaxAgeParts = responseControl
+      .find(value => value.startsWith("s-maxage"))
+      ?.split(/\s*=\s*/);
+    if (sMaxAgeParts.length > 1 && parseInt(sMaxAgeParts[0]) !== NaN) {
+      return parseInt(sMaxAgeParts[0]);
+    }
+
+    const maxAgeParts = responseControl
+      .find(value => value.startsWith("max-age"))
+      ?.split(/\s*=\s*/);
+    if (maxAgeParts.length > 1 && parseInt(maxAgeParts[0]) !== NaN) {
+      return parseInt(maxAgeParts[0]);
+    }
+  }
+
+  const responseExpires = Date.parse(response.headers.get("expires"));
+  const expiresInSeconds = Math.floor((responseExpires - Date.now()) / 1000);
+
+  return expiresInSeconds > 0 ? expiresInSeconds : 0;
+};
+
+/**
+ * Parses the URL string and sorts query params.
+ *
+ * @param urlString The URL to be normalized
+ */
+const normalizeUrl = (urlString: string): string => {
+  const parsed: url.URL = new url.URL(urlString);
+  parsed.searchParams.sort();
+  return parsed.toString();
+};
+
+/**
+ * Generate the cache key for the request to be cached or retrieved.
+ *
+ * @param req The request to generate a cache key for
+ */
+const makeCacheKey = (req: fetch.Request): string => normalizeUrl(req.url);
+
+const getMetadataKey = (req: fetch.Request): string =>
+  `request-cache-metadata:${makeCacheKey(req)}`;
+
+const getContentKey = (req: fetch.Request): string =>
+  `request-cache:${makeCacheKey(req)}`;
+
+const addCacheHeaders = (resHeaders, path, key, hash, time): void => {
+  resHeaders.set("X-Local-Cache", encodeURIComponent(path));
```

Comment:
I don't love this static list of headers.  But can live with it.

Reply:
Fixed